ffi/tests/ctests/src/lib.rs

```diff
@@ -0,0 +1,89 @@
+#[no_mangle]
+unsafe extern "C" fn rust_assert(
+    check: bool,
+    file: *const std::os::raw::c_char,
+    line: usize,
+    msg: *const std::os::raw::c_char,
+) {
+    assert!(
+        check,
+        "[{}:{}] {}",
+        std::ffi::CStr::from_ptr(file).to_str().unwrap(),
+        line,
+        std::ffi::CStr::from_ptr(msg).to_str().unwrap()
+    );
+}
+
+mod ffi_ctest {
+    #[test]
+    fn create_scheme() {
+        #[link(name = "wirefilter_ffi")]
```

Comment:
Maybe deduplicate these with some macro now that they're so similar?

Reply:
See my other comment about this.